crates/phactory/src/system/mod.rs

```diff
@@ -458,7 +475,16 @@ impl<Platform: pal::Platform> System<Platform> {
         let identity_key = WorkerIdentityKey(identity_key);
         let pubkey = identity_key.public();
         let sender = MessageOrigin::Worker(pubkey);
-        let master_key = master_key::try_unseal(sealing_path.clone(), &identity_key.0, &platform);
+        let master_key_history =
+            master_key::try_unseal(sealing_path.clone(), &identity_key.0, &platform);
+        let master_key = if master_key_history.len() == 0 {
+            None
+        } else {
+            let first_rotation = master_key_history.first().expect("checked; qed.");
```

Comment:
```suggestion
            let first_key = master_key_history.first().expect("checked; qed.");
```

It should be the key not rotated.

Reply:
This code has been refactored and removed.